XMonad/Prompt.hs

```diff
@@ -1227,6 +1229,23 @@ changeWord p = join $ f <$> getInput <*> getOffset <*> pure p
         f str off p'| p' $ str !! off       = killWord' (not . p') Next
                     | otherwise             = killWord' p' Next
 
+-- | Interchange characters around point, moving forward one character
+--   if not at the end of the input.
+transposeChars :: XP ()
+transposeChars = do
+  off <- gets offset
+  cmd <- gets command
+  let (beforeCursor, afterCursor) = splitAt off cmd
+      (ncom, noff) = fromMaybe (cmd, off) (go beforeCursor afterCursor off)
+  modify $ \s -> setCommand ncom $ s{ offset = noff }
+ where
+  go :: [a] -> [a] -> Int -> Maybe ([a], Int)
+  go (reverse -> (b1 : b2 : bs)) [] offset =  -- end of line
```

Comment:
I wonder why hlint doesn't complain about the extra parens here.

Reply:
(No need to remove them, it's just weird :-))

Reply:
hlint seems reluctant to suggest this in patterns, it seems.  The following are also completely fine (on `HLint v3.3.1`): 

``` haskell
f :: [a] -> [a]
f ((((a : as)))) = as
f _              = []

g :: [a] -> [a]
g = \case
  ((((a:as)))) -> as
  _            -> []
```